### PR TITLE
[MNOE-455] Impersonation - Add user consent

### PR DESCRIPTION
**Pull request by x4d3**

### CODE REVIEW

File: src/app/components/mnoe-config/mnoe-admin-config.svc.coffee [hunk 1]
```diff
@@ -26,6 +26,13 @@
     else
       true
 
+  @isImpersonationConsentRequired = () ->
+    if ADMIN_PANEL_CONFIG.impersonation?.consent_required?
+      ADMIN_PANEL_CONFIG.impersonation.consent_required
+    else
+      false
+
+
```
Comment (alexnoox) [inline src/app/components/mnoe-config/mnoe-admin-config.svc.coffee]: Remove random double empty spaces, one is enough.

File: src/app/components/mnoe-users-local-list/mnoe-users-local-list.coffee [hunk 2]
```diff
@@ -17,6 +17,8 @@
       search: ''
 
     scope.isImpersonationEnabled = MnoeAdminConfig.isImpersonationEnabled()
+    scope.isImpersonationConsentRequired = MnoeAdminConfig.isImpersonationConsentRequired()
+
```
Comment (alexnoox) [inline src/app/components/mnoe-users-local-list/mnoe-users-local-list.coffee]: Remove random double empty spaces, one is enough.

File: src/app/components/mnoe-impersonate-button/mnoe-impersonate-button.coffee [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    vm.isImpersonationEnabled = MnoeAdminConfig.isImpersonationEnabled()
+
+    vm.impersonationStatus = ->
+      if MnoeAdminConfig.isImpersonationConsentRequired()
+        switch vm.user.access_request_status
+          when 'approved' then 'approved'
+          when 'never_requested', 'expired', 'denied' then 'requestable'
+          when 'requested' then 'requested'
+      else
+        'approved'
```
Comment (alexnoox) [inline src/app/components/mnoe-impersonate-button/mnoe-impersonate-button.coffee]: For those labels, maybe we could do something like this (cf. https://github.com/alexnoox/mnoe-admin-panel/blob/1.0/src/app/index.constants.coffee):
```
.constant('USER_ROLES', [
    {value: 'Member', label: 'mnoe_admin_panel.constants.user_roles.member'},
    {value: 'Admin', label: 'mnoe_admin_panel.constants.user_roles.admin'},
    {value: 'Super Admin', label: 'mnoe_admin_panel.constants.user_roles.super_admin'}
  ])
```

Comment (x4d3) [inline src/app/components/mnoe-impersonate-button/mnoe-impersonate-button.coffee]: Here they are not label, but code used in the switch

**Comment by ouranos** [comment]
Ready to merge once backend merged

**Pull request merged**